db/python/layers/analysis.py

```diff
@@ -132,6 +138,80 @@ async def query_analysis(
 
         return analyses
 
+    async def get_sample_file_sizes(
+        self,
+        project_ids: List[int] = None,
+        sample_ids: List[int] = None,
+        start_date: date = None,  # pylint: disable=W0613
+        end_date: date = None,  # pylint: disable=W0613
+    ):
+        """
+        Get the file sizes from all the given projects group by sample filtered
+        on the date range
+        """
+
+        projects = dict(zip(sample_ids, project_ids))
+
+        # Get sample history
+        history = await self.sampt.get_sample_create_date(sample_ids)
+
+        def keep_sample(sid):
+            d = history[sid]
+            if start_date and d <= start_date:
+                return True
+            if end_date and d <= end_date:
+                return True
+            if not start_date and not end_date:
+                return True
+            return False
+
+        # Get size of analysis crams
+        use_samples = list(filter(keep_sample, sample_ids))
+        crams = await self.at.query_analysis(
+            sample_ids=use_samples,
+            analysis_type=AnalysisType.CRAM,
+            status=AnalysisStatus.COMPLETED,
+        )
+        crams_by_sid: dict[int, dict[SequenceType, list]] = defaultdict(
+            lambda: defaultdict(list)
+        )
+
+        # Manual filtering to find the most recent analysis cram of each sequence type
+        # for each sample
+        for cram in crams:
+            sids = cram.sample_ids
+            seqtype = cram.meta.get('sequence_type')
+            seqtype = seqtype if seqtype else cram.meta.get('sequencing_type')
+            size = cram.meta.get('size')
+
+            if len(sids) == 1 and not isinstance(seqtype, list) and seqtype and size:
```

Comment:
Should raise an exception if this isn't the case (except maybe seqtype`).

Reply:
But aren't there analyses that have multiple sample ids in the db? If that's the case I was simply going to ignore them quietly since there is no straight forward way to determine the file sizes.

Reply:
But you're filtering for crams only - there shouldn't be any crams with more than one sample ID right? I guess technically possible, but If that happens, sounds like we've made a mistake. For now could this script collect all the cases it encounters, and logs them once after the for loop.

If possible to turn this into an early return, eg:

```python
if len(sids) > 1:
    affected_analyses.append(cram['id')
    continue
# other stuff
```